tests/DependencyEmitter/UsesDependencyEmitterTest.php

```diff
@@ -59,4 +59,19 @@ public function testIncludesFQDNWhichIsAlsoANamespacePrefix(): void
         self::assertCount(1, $deps);
         self::assertContains('FQDNNamespacePrefix\Uses\Foo:5 on FQDNNamespacePrefix\FQDN', $deps);
     }
+
+    public function testIncludesFQDNWhichIsDoesNotExistAndSubstringOfAnotherFQDN(): void
```

Comment:
Here is a case that shows the difference between main and my PR. This test will fail on main branch.